apps/api/src/app/events/usecases/trigger-event/message-matcher.service.ts

```diff
@@ -0,0 +1,266 @@
+    );
+
+    return { asyncFilters, syncFilters };
+  }
+  private async handleAndFilters(filter, variables: IFilterVariables, configuration: IMessageFilterConfiguration) {
+    const { asyncFilters, syncFilters } = this.splitToSyncAsync(filter);
+
+    const foundSyncFilterMatches = syncFilters.filter((i) => this.processFilterEquality(variables, i));
+    if (syncFilters.length !== foundSyncFilterMatches.length) {
+      return false;
+    }
+
+    const foundAsyncFilterMatches = await filterAsync(asyncFilters, (i) =>
+      this.processFilter(variables, i, configuration)
+    );
+
+    return foundAsyncFilterMatches.length === asyncFilters.length;
+  }
+
+  private async handleOrFilters(filter, variables: IFilterVariables, configuration: IMessageFilterConfiguration) {
+    const { asyncFilters, syncFilters } = this.splitToSyncAsync(filter);
+
+    const syncRes = syncFilters.find((i) => this.processFilterEquality(variables, i));
+    if (syncRes) {
+      return true;
+    }
+
+    return await findAsync(asyncFilters, (i) => this.processFilter(variables, i, configuration));
+  }
+
+  private processFilterEquality(variables: IFilterVariables, i) {
+    const payloadVariable = _.get(variables, [i.on, i.field]);
+    const value = this.parseValue(payloadVariable, i.value);
+    if (i.operator === 'EQUAL') {
+      return payloadVariable === value;
+    }
+    if (i.operator === 'NOT_EQUAL') {
+      return payloadVariable !== value;
+    }
+    if (i.operator === 'LARGER') {
+      return payloadVariable > value;
+    }
+    if (i.operator === 'SMALLER') {
+      return payloadVariable < value;
+    }
+    if (i.operator === 'LARGER_EQUAL') {
+      return payloadVariable >= value;
+    }
+    if (i.operator === 'SMALLER_EQUAL') {
+      return payloadVariable <= value;
+    }
+    if (i.operator === 'NOT_IN') {
+      return !payloadVariable.includes(value);
+    }
+    if (i.operator === 'IN') {
+      return payloadVariable.includes(value);
+    }
+
+    return false;
+  }
+
+  private async getWebhookResponse(
+    child,
+    variables: IFilterVariables,
+    configuration: IMessageFilterConfiguration
+  ): Promise<Record<string, unknown>> {
+    if (!child.webhookUrl) return undefined;
+
+    const payload = await this.buildPayload(variables, configuration);
+
+    const hmac = await this.buildHmac(configuration);
+
+    const config = {
+      headers: {
+        'nv-hmac-256': hmac,
+      },
+    };
+
+    try {
+      return await axios.post(child.webhookUrl, payload, config).then((response) => {
+        return response.data as Record<string, unknown>;
+      });
+    } catch (err) {
+      throw new Error(
+        JSON.stringify({
+          message: err.message,
+          data: EXCEPTION_MESSAGE_ON_WEBHOOK_FILTER,
+        })
+      );
+    }
+  }
+
+  private async buildPayload(variables: IFilterVariables, configuration: IMessageFilterConfiguration) {
+    if (process.env.NODE_ENV === 'test' && !configuration) return variables;
```

Comment:
@djabarovgeorge why this is needed here? if it's really needed maybe worth introducing a variable os it will be self-explained or a comment?